test/functorch/test_control_flow.py

```diff
@@ -1429,9 +1463,11 @@ def f_wrapper(func):
             def wrapper(*args, **kwargs):
                 torch._enable_functionalization(reapply_views=False)
                 try:
-                    func_args = pytree.tree_map(to_fun, args)
-                    func_kwargs = pytree.tree_map(to_fun, kwargs)
-                    return pytree.tree_map(from_fun, func(*func_args, **func_kwargs))
+                    func_args = pytree.tree_map(
+                        lambda x: to_fun_old(x) if isinstance(x, torch.Tensor) else x, args)
+                    func_kwargs = pytree.tree_map(
```

Comment:
TBH if we do the refactor into FunctorchFunctionalizeAPI, CppFunctionalizeAPI, PythonFunctionalizeAPI, we could use those as the sources of truth and just call CppFunctionalizeAPI.unwrap_tensors(...)